image-hd.c

```diff
@@ -125,13 +125,20 @@ static void lba_to_chs(unsigned int lba, unsigned char *chs)
 	chs[2] = (c & 0xff);
 }
 
-static void hdimage_setup_chs(struct mbr_partition_entry *entry)
+static void hdimage_setup_chs_with_offset(struct mbr_partition_entry *entry,
+					  unsigned long long offset_sectors)
 {
-	lba_to_chs(entry->relative_sectors, entry->first_chs);
-	lba_to_chs(entry->relative_sectors + entry->total_sectors - 1,
+	lba_to_chs(entry->relative_sectors + offset_sectors,
+		   entry->first_chs);
+	lba_to_chs(entry->relative_sectors + entry->total_sectors - 1 + offset_sectors,
 		   entry->last_chs);
 }
 
+static void hdimage_setup_chs(struct mbr_partition_entry *entry)
+{
+	hdimage_setup_chs_with_offset(entry, 0);
+}
```

Comment:
No separate function, just call it with offset zero in `hdimage_setup_mbr`.

Reply:
Sure, adjusted.